### PR TITLE
firmware: added rpl doc

### LINKED ISSUE #243: Doc: need to add documentation of some modules
#### Description

It is necessary to add the documentation of the following modules:
- [ ] rpl_protocol
- [ ] radio
### Useful links

* https://doc.riot-os.org/group__drivers__netdev__ieee802154.html#gaec403a7dcf6581e11b95b75220c40a35
* https://doc.riot-os.org/group__net__gnrc__rpl.html
* https://doc.riot-os.org/group__net__gnrc__rpl__srh.html

### PR DESCRIPTION
**Pull request by xkevin190**
### Contribution description

the documentation of the rpl protocol was added as well as the modules that were added for the use of mesh4all as a brief explanation of its operation.

### Issues/PRs references

this PR fixes #243

### CODE REVIEW

**Review by eduazocar — CHANGES_REQUESTED** [review]
Looks good for me, do squash

File: firmware/network/rpl_protocol/doc.txt [hunk 1]
```diff
@@ -4,16 +4,16 @@
 
 ### What goes here?
 
-In this module you will find the functions to start rpl and add a node dodag (Raiz) 
+In this module you will find the functions to start rpl and add a node dodag (Raiz)
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:7]: did you mean (Root)?

File: firmware/network/rpl_protocol/doc.txt [hunk 2]
```diff
@@ -4,16 +4,16 @@
...
 or dag (Slave) first rpl is initialized then it is verified if the node is dodag or dag and
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:8]: then it verifies ...

File: firmware/network/rpl_protocol/doc.txt [hunk 3]
```diff
@@ -4,16 +4,16 @@
...
-they are started an ipv6 is added where the dodag will be hosted one Once the dodag is started, 
+they are started an ipv6 is added where the dodag will be hosted one Once the dodag is started,
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:9]: if they are started, an ipv6 address is added where the dodag will be hosted once the dodag is initialized,

**Review by RocioRojas — CHANGES_REQUESTED** [review]

File: firmware/network/rpl_protocol/doc.txt [hunk 4]
```diff
@@ -4,16 +4,16 @@
...
 the DAG riot node is in charge of executing the entire routing process.
 
 ### How's work RPL protocol
 
 the RPL protocol is quite complex in this document we will try to explain it as simple to understand as possible.
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:14]: The RPL ...*

File: firmware/network/rpl_protocol/doc.txt (hunk 1 above)
Comment (RocioRojas) [reply firmware/network/rpl_protocol/doc.txt:7]: RPL* capitalized

File: firmware/network/rpl_protocol/doc.txt (hunk 2 above)
Comment (RocioRojas) [reply firmware/network/rpl_protocol/doc.txt:8]: RPL ... capitalized

File: firmware/network/rpl_protocol/doc.txt [hunk 5]
```diff
@@ -4,16 +4,16 @@
 
 ### What goes here?
 
-In this module you will find the functions to start rpl and add a node dodag (Raiz) 
+In this module you will find the functions to start rpl and add a node dodag (Raiz)
 or dag (Slave) first rpl is initialized then it is verified if the node is dodag or dag and
-they are started an ipv6 is added where the dodag will be hosted one Once the dodag is started, 
+they are started an ipv6 is added where the dodag will be hosted one Once the dodag is started,
 the DAG riot node is in charge of executing the entire routing process.
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:10]: Riot* capitalized

File: firmware/network/rpl_protocol/doc.txt [hunk 6]
```diff
@@ -25,10 +25,9 @@ the message is unicast by the child to the selected parent(s).
 `DIO` each node transmits DIO messages using the trickle timer. The frequency of DIO messages depends on network stability.
 
 As the generation of these messages incur additional energy consumption, RPL uses the Trickle algorithm to dynamically adjust the transmission
-The DIO message originates from the sink while the DAO originates from the nodes. This trickle timer helps keep track of the messages sent by the 
+The DIO message originates from the sink while the DAO originates from the nodes. This trickle timer helps keep track of the messages sent by the
 sink when it tells the nodes “I am the sink”. In RPL the DIO’s are sent out periodically and the time gap is increased after every transmission. This is done with the help of a trickle timer.
 
-
 for more information about the rpl protocol you can check the official documentation of the protocol [here](https://datatracker.ietf.org/doc/html/rfc6550)
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:31]: RPL* capitalized

**Review by RocioRojas — APPROVED** [review]

File: firmware/network/rpl_protocol/doc.txt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+the message is unicast by the child to the selected parent(s).
+
+`DIO` each node transmits DIO messages using the trickle timer. The frequency of DIO messages depends on network stability.
+
+As the generation of these messages incur additional energy consumption, RPL uses the Trickle algorithm to dynamically adjust the transmission
+The DIO message originates from the sink while the DAO originates from the nodes. This trickle timer helps keep track of the messages sent by the
+sink when it tells the nodes “I am the sink”. In RPL the DIO’s are sent out periodically and the time gap is increased after every transmission. This is done with the help of a trickle timer.
+
+
+This graph will show us how rpl works and the control messages.
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:34]: RPL* Capitalized

File: firmware/network/rpl_protocol/doc.txt [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+\image html rpl.png RPL messages width=40%
+
+for more information about the RPL protocol you can check the official documentation of the protocol [here](https://www.rfc-editor.org/rfc/rfc6550.txt)
+
+
+In our investigation we were able to find a lot of prl documentation that helped us understand the correct functioning of this, among them are:
```
Comment (RocioRojas) [inline firmware/network/rpl_protocol/doc.txt:41]: lot of RPL*....

Comment (xkevin190) [reply firmware/network/rpl_protocol/doc.txt:41]: what do you mean?

Comment (RocioRojas) [reply firmware/network/rpl_protocol/doc.txt:41]: you write a lot of prl, maybe you want to write "a lot of RLP*"

Comment (eduazocar) [reply firmware/network/rpl_protocol/doc.txt:41]: a lot of RPL documentation*

**Review by eduazocar — APPROVED** [review]

**Pull request merged**